aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -271,6 +307,12 @@ void OperatorEntry::checkInvariants() const {
     dispatchKeyExtractor().checkInvariants(schema_->schema);
   }
   TORCH_INTERNAL_ASSERT(kernels_.find(DispatchKey::Undefined) == kernels_.end(), dumpState());
+  TORCH_INTERNAL_ASSERT(kernels_.find(DispatchKey::Math) == kernels_.end()
+    || !hasKernelForDispatchKeySet(getBackendKeySetFromAutograd(DispatchKey::AutogradOther)),
+    "You're registering kernel to a backend maps to AutogradOther trying to override Math kernel behavior. "
+    "If this is intended, please open an issue to request a proper Autograd dispatch key for your backend.",
+    dumpState()
+  );
```

Comment:
This doesn't look right. In general you can't expect `checkInvariants` will be called. If you want a user facing message, you should report the error elsewhere; maybe in dispatch table recalcluation?

Reply:
Aha I'll create a new kernel as you suggested above and put the error message there ;)